### PR TITLE
Added pure websocket support... enabling us to drop sockjs

### PR DESCRIPTION
**Pull request by jonasfj**
Added proxy thing that uses websocket...

It's hard to test all the corner cases here... So please just given a quick glance and tell me if you think I'm crazy... :)

I've tried to be fairly robust and make sure that a stupid client doesn't leak resources or causes other clients to disconnect... Granted ddos'ing events.tc.net is always an option. Perhaps we need some per IP limitation...
Or maybe we do that if it becomes a problem (I think not).

Note, no auth is needed to use this, which is also why queue is max 50, you can only have one queue per websocket (browser usually limit number of websockets per site). However, pulse is protected in the way that this is intended to crash before we overload pulse.

Once this is landed... I'll develop a client side component that uses websocket...

### CODE REVIEW

File: events/websocket.js [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+    this.socket.ping(slugid.v4());
+  }
+  catch (err) {
+    debug("Failed to send ping");
+    this.close();
+  }
+};
+
+/** Handle pongs, so that keepAlive won't close() in next iteration */
+Proxy.prototype.onPong = function(data, flags) {
```
Comment (lightsofapollo) [inline events/websocket.js:387]: It is not particularly clear here why you want to do the ping/pong can you add a longer comment describing why?

Comment (jonasfj) [inline events/websocket.js:387]: Basically, we need to send something to ensure that the TCP connection is still alive... Technically, we don't have to get something as missing acks will close the TCP connection, but it's nice get pongs... Longer comment added...

**Comment by jonasfj** [comment]
I think this was reviewed by jhford a while ago... I just haven't landed it as fixing this was down prioritized..

**Comment by jonasfj** [comment]
Mostly we need to fix the client before rolling this out has any value.

**Comment by petemoore** [comment]
@jonasfj is this ready to roll out? Thanks!

**Comment by jonasfj** [comment]
Not sure I ever figured out the ping pong...

It should probably be ready to merfe, but we don't have a client for it
yet...
Den 7. feb. 2016 11.51 PM skrev "Pete Moore" notifications@github.com:

**Comment by gregarndt** [comment]
considering this PR hasn't been updated in over a year, should we just shelve this idea and close out the PR?

**Comment by jonasfj** [comment]
@gregarndt, another option is to leave it open, that way we'll find it... should come back aiming to fix some day?
